packages/SystemUI/src/com/android/systemui/biometrics/UdfpsController.java

```diff
@@ -1188,10 +1233,17 @@ private void onFingerDown(
                 view.configureDisplay(() -> dispatchOnUiReady(requestId));
             }
         }
+        if(SystemProperties.get("ro.vendor.build.fingerprint").contains("nubia/NX669")) {
+          processCmd(13, 0, 0, new byte[0], 0);
+        }
 
         for (Callback cb : mCallbacks) {
             cb.onFingerDown();
         }
+        if(SystemProperties.get("ro.vendor.build.fingerprint").contains("nubia/NX669")) {
```

Comment:
There should be either some timing or some call back for that 14 event, since that event is supposed to say "hey the green/white circle with HBM is displayed" which should take few frames. Having this timing/callback would probably improve the responsiveness of the FOD.

That being said, I can merge it as-is if you're satisfied with it

Reply:
No need to delay for fingerprint white/green circle. After sending the cmd "13" sensor takes 25-200ms to scan the fingerprint. It is enough time for HBM active. In the source code, it only delays 10ms for HBM to be active. It never fails to scan fp for me.